src/frontend/js/components/CourseAddToWishlist/index.tsx

```diff
@@ -0,0 +1,128 @@
+import { useEffect, useState } from 'react';
+import { defineMessages, FormattedMessage } from 'react-intl';
+import { useUserWishlistCourses } from 'hooks/useUserWishlistCourses';
+import type * as Joanie from 'types/Joanie';
+import { Button } from 'components/Button';
+import { Spinner } from 'components/Spinner';
+import { useSession } from 'data/SessionProvider';
+
+enum CourseAddToWishlistMessageIds {
+  LABEL_ADD = 'labelAdd',
+  LABEL_REMOVE = 'labelRemove',
+  ERROR_ADD = 'errorAdd',
+  ERROR_REMOVE = 'errorRemove',
+  LOG_ME = 'logMe',
+}
+const messages = defineMessages({
+  [CourseAddToWishlistMessageIds.LABEL_ADD]: {
+    id: 'components.CourseAddToWishlist.labelAdd',
+    description: 'label to proceed to add a course to user wishlist',
+    defaultMessage: 'Notify me',
+  },
+  [CourseAddToWishlistMessageIds.LABEL_REMOVE]: {
+    id: 'components.CourseAddToWishlist.labelRemove',
+    description: 'label to proceed to remove a course to user wishlist',
+    defaultMessage: 'Do not notify me',
+  },
+  [CourseAddToWishlistMessageIds.ERROR_ADD]: {
+    id: 'components.CourseAddToWishlist.errorAdd',
+    defaultMessage:
+      'An error occurred when adding this course to your wishlist. Please retry later.',
+    description: 'Error message shown when add to wishlist request failed.',
+  },
+  [CourseAddToWishlistMessageIds.ERROR_REMOVE]: {
+    id: 'components.CourseAddToWishlist.errorRemove',
+    defaultMessage:
+      'An error occurred when removing this course to your wishlist. Please retry later.',
+    description: 'Error message shown when remove to wishlist request failed.',
+  },
+  [CourseAddToWishlistMessageIds.LOG_ME]: {
+    id: 'components.CourseAddToWishlist.logMe',
+    defaultMessage: 'Log in to be notify',
+    description: 'label to proceed to login page before beeing adding a course to user wishlist.',
+  },
+});
+
+enum ComponentStates {
+  INITIALIZING = 'initializing',
+  LOADING = 'loading',
+  IDLE = 'idle',
+  ERROR = 'error',
+}
+
+export interface Props {
+  courseCode: Joanie.Course['code'];
+}
+
+const CourseAddToWishlist = ({ courseCode }: Props) => {
+  const [componentState, setComponentState] = useState<ComponentStates>(
+    ComponentStates.INITIALIZING,
+  );
+  const [isInWishlist, setIsInWishlist] = useState<boolean>(false);
```

Comment:
Do we really need those states? I think we can prevent some extra rendering.

I think component state can be guessed through `useUserWishlistCourses`.
```ts
const componentState = useMemo(() => {
  if (wishlistCourse.fetching) return ComponentStates.LOADING;
  if (wishlistCourse.errors) return ComponentStates.ERROR;
  return ComponentStates.IDLE;
}, [wishlistCourse.errors, wishlistCourse.fetching]); 
```

Idem for isInWishList which can be a variable :
```ts
const isInWishListCourse = useMemo(() => wishlistCourse.items?.length > 0, wishlistCourse.items)
```